src/Pages/SingleVideoPage/SingleVidePageComponents/VideoInfo/VideoInfo.jsx

```diff
@@ -0,0 +1,58 @@
+import { Link, useParams } from "react-router-dom";
+import {
+  useAuthContext,
+  usePlaylistContext,
+  useReducerContext,
+} from "../../../../Context";
+import "./VideoInfo.css";
+
+export const VideoInfo = () => {
+  const { videoId } = useParams();
+  const { videos } = useReducerContext();
+  const { userState } = useAuthContext();
+  const { setPlaylistModal } = usePlaylistContext();
+
+  return videos
+    .filter((video) => video._id === videoId)
+    .map((video) => {
+      return (
+        <div key={video._id} className="videoInfo">
+          <h1>{video.title}</h1>
+          <div className="video-views">
+            <h2>
+              {video.views} | {video.publishDate}
+            </h2>
+            <div className="btn-container">
+              <div className="like">
+                <span className="material-icons">thumb_up_alt</span>
+                <span className="text">Like</span>
+              </div>
+              <div className="watch">
+                <span className="material-icons">watch_later</span>
+                <span className="text">Watch Later</span>
+              </div>
+
+              {Object.keys(userState).length === 0 ? (
+                <Link to="/login">
+                  <div className="add-to-playlist">
+                    <span className="material-icons">queue</span>
+                    <span className="text">Add to Playlist</span>
+                  </div>
+                </Link>
+              ) : (
+                <div
+                  onClick={() =>
+                    setPlaylistModal({ state: true, video: video })
+                  }
+                  className="add-to-playlist"
+                >
+                  <span className="material-icons">queue</span>
+                  <span className="text">Add to Playlist</span>
+                </div>
+              )}
+            </div>
+          </div>
+        </div>
```

Comment:
Is there any specific reason why video description and video info are put into two separate components ? They share almost all the declarations and filtering + mapping of data. I suppose combining them together would mean less code and more reuse.

Reply:
As the videoInfo component holds like, watch later and add to playlist buttons jsx will increase overtime for that component.
taking that consideration in mind i decided to create a new component for description and info seperatly.
At start these 2 were single component but it was looking a little stuffed so i created different components.